src/domain/user/services/ListService.ts

```diff
@@ -1,20 +1,28 @@
-import User from '../mocks/UserMock'
-import Iuser from '../types/UserTypes'
+import IListService from '../../../interfaces/IListService'
+import IUser from '../../../interfaces/IUser'
+import IServiceResponse from '../../../interfaces/IServiceResponse'
+import { inject, injectable } from 'tsyringe'
+import IUserRepository from '../../../interfaces/IUserRepository'
 
-class ListService {
- static async listUser() {
-  try {
-   const list: Partial<Iuser>[] = []
-   for (const pessoa of User) {
-    const { fullName, email, birthdate } = pessoa
+@injectable()
+export default class ListService implements IListService {
+  userRepository: IUserRepository
+  constructor (@inject('UserRepository')userRepository: IUserRepository) {
+    this.userRepository = userRepository
+  }
 
-    list.push({ fullName, email, birthdate })
-   }
-   return { code: 201, msg: list }
-  } catch (error) {
-   return { code: 422, msg: error }
+  async listUser (): Promise<IServiceResponse> {
+    try {
+      const list: Partial<IUser>[] = this.userRepository.database.map((item) => {
+        const { fullName, email } = item
+        return {
+          fullName,
+          email
+        }
+      })
+      return { code: 201, msg: list }
+    } catch (error) {
+      return { code: 422, msg: error }
+    }
```

Comment:
1. Recomendo dar uma olhada no artigo https://thorben-janssen.com/dont-expose-entities-in-api/ do por que não devemos mesma entidade do domínio para api
2. Não recomendo usar Partial pois pode causar uma grande confusão na typagem do código(caso real aqui da Warren)
3. 
Recomendo criar interface IUserResponse para utilizar como retorno da api
```suggestion
  async listUser (): Promise<IServiceResponse> {
    try {
      const result: IUserResponse[] = this.userRepository.database.map({ fullName, email } => {
        return {
          fullName,
          email
        }
      })
      return { code: 201, msg: result }
    } catch (error: any) {
      return { code: 422, msg: error.message }
    }
```